specs/snapshots.md

```diff
@@ -0,0 +1,91 @@
+- `App::commit` calls `SnapshotClient::notify` with the current block height and the committed state parameters.
+- `tmconv::to_snapshot` maps the fields of a `SnapshotManifest` to the CometBFT type, in particular it encodes the `FvmStateParams` into the `metadata` field
+- `tmconv::from_snapshot` tries to restore a `SnapshotManifest` from an offer, comparing the CID of the decoded `FvmStateParams` (which is the `AppHash` of Fendermint) to the `app_hash` in the offer, thus checking the integrity that the offer is coming from a remote which, according to the Light Client, has a consensus that agrees with the snapshot (unless they lie about the snapshot contents).
+
+<aside>
+💡 There are some comments in `App::commit` regarding how the state is related to block height. Remember that state which is the result of executing block `H` is only available to query at height `H+1`, which is reflected in `AppState::state_height` that we use to determine at which height the state is inserted into the state history.
+
+When we do snapshots, however, we have to use height `H` with the state parameters that we save at `H+1`. The reason for this is because in `offer_snapshot` containing the snapshot we produced for `H` CometBFT will actually give us the `app_hash` at `H+1`, and this way things line up for validation.
+
+In other words, a CometBFT `Snapshot` for height `H` contains the state which is the result of the execution of block `H`, which CometBFT allows to query with height `H+1`.
+
+</aside>
+
+# Snapshot Configuration
+
+Snapshots are not enabled by default. To enabling them requires configuration in two places:
+
+- in the [Fendermint config](https://github.com/consensus-shipyard/ipc/blob/7af25c4c860f5ab828e8177927a0f8b6b7a7cc74/fendermint/app/config/default.toml#L68), e.g. by setting the `FMT_SNAPSHOT__ENABLED` env var
+- in the [CometBFT config](https://docs.cometbft.com/v0.37/core/configuration) by setting multiple variables:
+    - `CMT_STATESYNC_ENABLE` to allow CometBFT to bootstrap from others’ snapshots
+    - `CMT_STATESYNC_RPC_SERVERS` with a list of at least 2 CometBFT peers where the Light Client can go and catch up with the state of the chain before CometBFT starts asking for snapshots
+    - `CMT_STATESYNC_TRUST_HEIGHT` is a height of a recent block which the operator knows to be on the correct chain
+    - `CMT_STATESYNC_TRUST_HASH` is the hash of that block
+
+See the documentation of [State Sync](https://docs.cometbft.com/v0.37/core/state-sync) for how to obtain such trusted values.
+
+The recommended practice by Tendermint is never to trust a block header older than the unbonding period of the Proof-of-Stake consensus, to protect against long range attacks. Within the unbonding period, if evidence of an equivocation arises, the offending validators can still be slashed, but outside that period, there is much less that can be done. The default trust period in CometBFT is 1 week.
```

Comment:
This is a useful fact to note in the spec portion, not the configuration section.